src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6422,7 +6423,8 @@ private ScanTypeFlags ScanType(ParseTypeMode mode, out SyntaxToken lastTokenOfTy
                 {
                     case SyntaxKind.QuestionToken
                             when lastTokenOfType.Kind != SyntaxKind.QuestionToken && // don't allow `Type??`
-                                 lastTokenOfType.Kind != SyntaxKind.AsteriskToken: // don't allow `Type*?`
+                                 lastTokenOfType.Kind != SyntaxKind.AsteriskToken && // don't allow `Type*?`
+                                 !isFunctionPointer: // don't allow `delegate*<...>?`
```

Comment:
>// don't allow `delegate*<...>?` [](http://example.com/codeflow?start=53&length=32)

Why not? It's unambiguous what it means. It's _bad_, true, but unambiguous. I think it would make error recovery better.

Reply:
The behavior of `ScanType()` is different today from `ParseType()` for this case, making it difficult to recover from cases such as `delegate*<void>? () => x`.

> Why not? It's unambiguous what it means. It's bad, true, but unambiguous. I think it would make error recovery better.

Is `delegate*<...>?` different in that regard from `Type??` or `Type*?` which we disallow here?

Reply:
> Is delegate*<...>? different in that regard from Type?? or Type*? which we disallow here?

I think so, yes. In either of those cases, `Type` is not guaranteed to _definitely_ be a type. `delegate*` is 100% a type and cannot be anything else.

Reply:
> The behavior of ScanType() is different today from ParseType() for this case, making it difficult to recover from cases such as delegate*<void>? () => x.

I'm not opposed to making the behavior of ParseType be the same here, parsing into an invalid nullable type.

Reply:
i agree with fred here.  i'd far prefer that we parse these out as types (unless htat would be bad because it woudl actually break some level ternary-expression case).  And then later on have binding/whatever error and say that a nullable func-ptr is not legal.